lti_consumer/lti_1p3/tests/test_consumer.py

```diff
@@ -726,3 +727,37 @@ def test_set_dl_content_launch_parameters(self):
             {"test": "test"}
         )
         self.assertEqual(self.lti_consumer.launch_url, "example.com")
+
+    def test_no_nrps_returns_failure(self):
+        """
+        Test that when LTI NRPS isn't configured, the class yields an error.
+        """
+        with self.assertRaises(exceptions.LtiNRPSServiceNotSetUp):
+            self.lti_consumer.lti_nrps  # pylint: disable=pointless-statement
+
+    def test_enable_nrps(self):
+        """
+        Test enabling LTI NRPS and checking that required parameters are set.
+        """
+        self.lti_consumer.enable_nrps("http://example.com/20/membership")
+
+        # Check that the NRPS class was properly instanced and set
+        self.assertEqual(type(self.lti_consumer.nrps), LtiNrps)
+
+        # Check retrieving class works
+        lti_nrps_class = self.lti_consumer.lti_nrps
+        self.assertEqual(self.lti_consumer.nrps, lti_nrps_class)
```

Comment:
Is this testing an internal detail of the class?  Is "nrps" private or not?

Reply:
`nrps` is not supposed to be used directly but through `lti_nrps`. This tests if `lti_nrps` returns the same object as `nrps` attribute.